docs/advanced/classes.rst

```diff
@@ -559,6 +559,39 @@ crucial that instances are deallocated on the C++ side to avoid memory leaks.
     py::class_<MyClass, std::unique_ptr<MyClass, py::nodelete>>(m, "MyClass")
         .def(py::init<>())
 
+.. _destructors_that_call_python:
+
+Destructors that call Python
+============================
+
+If a Python function is invoked from a C++ destructor, an exception may be thrown
+of type :class:`error_already_set`. If this error is thrown out of a class destructor,
```

Comment:
Since this is classes.rst I was keeping this focused on destructors and letting the exceptions section address both.

While destructors can throw, I believe that `py::~object` being implicitly `noexcept(true)` means no pybind11 object of interest can have a throwing destructor. I tried a throwing destructor once - it crashes - didn't explore much further. I think the situation is that throwing destructors are not supported unless someone decides to trying supporting them.

Reply:
Riiiight, OK. Good point; I hadn't considered inheritance!